src/views/Profile/components/EditProfileModal/reducer.ts

```diff
@@ -0,0 +1,66 @@
+import { useReducer } from 'react'
```

Comment:
A note on this - I thought it was going to be much more complicated so I added a reducer but in the end it is rather simple.